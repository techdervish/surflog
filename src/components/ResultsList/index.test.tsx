import {
  createShallowWrapper,
  expectWrappersTreeToMatchSnapshot,
  mocks,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('ResultsList component', () => {
  const wrapper = createWrapper({
    items: mocks.searchResults,
    isFirstSearchDone: true,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('renders correct number of item', () => {
    expect(wrapper.find('ListItem')).toHaveLength(mocks.searchResults.length);
  });

  it('renders no record when items count is 0 and is first search done', () => {
    const noItemWrapper = createWrapper({ items: [], isFirstSearchDone: true });
    expect(noItemWrapper.find('div').text()).toEqual('No result found');
  });

  it('renders nothing when items count is 0 and is first search not done', () => {
    const noItemWrapper = createWrapper({
      items: [],
      isFirstSearchDone: false,
    });
    expect(noItemWrapper.find('div').text()).toEqual('');
  });
});
