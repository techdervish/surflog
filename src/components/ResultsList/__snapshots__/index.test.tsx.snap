// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList component renders the expected snapshot 1`] = `
<div
  style={
    Object {
      "fontSize": 14,
      "margin": "0 -16px",
      "overflowWrap": "break-word",
    }
  }
>
  <ListItem
    key="0_item"
    occurences={
      Array [
        Object {
          "key": "content",
          "position": Array [
            61,
            5,
          ],
        },
      ]
    }
    onClick={[Function]}
    record={
      Object {
        "content": "·8mReplying to @BenLeshI'm mildly surprised I haven't had to build @ladyleet a fort yet. 1",
        "id": "/BenLesh/status/1160394371784744962",
        "lastVisitedTime": 1565495077688,
        "link": "https://twitter.com//BenLesh/status/1160394371784744962",
        "page": "twitter",
        "title": "Ben Lesh@BenLesh",
      }
    }
  />
  <ListItem
    key="1_item"
    occurences={
      Array [
        Object {
          "key": "content",
          "position": Array [
            17,
            5,
          ],
        },
      ]
    }
    onClick={[Function]}
    record={
      Object {
        "content": "·16mI decided to build a box fort for my 4 yr old out of some of our moving boxes... And maybe it got out of hand. 623",
        "id": "/BenLesh/status/1160392474503897091",
        "lastVisitedTime": 1565495072153,
        "link": "https://twitter.com//BenLesh/status/1160392474503897091",
        "page": "twitter",
        "title": "Ben Lesh@BenLesh",
      }
    }
  />
</div>
`;
